src/edu/csus/ecs/pc2/ui/ShadowCompareScoreboardPane.java

```diff
@@ -0,0 +1,545 @@
+                String numSolved = getModel().getValueAt(modelRow, ScoreboardColumnId.NUM_SOLVED.ordinal()).toString();
+                String time = getModel().getValueAt(modelRow, ScoreboardColumnId.TOTAL_TIME.ordinal()).toString();
+                
+                //assume there IS a corresponding row in the other scoreboard table
+                boolean hasCorrespondingRow = true;
+                
+                //check whether the specified row has all data values set to the string "?" -- if so, there's no corresponding
+                // row in the other table.  (See {@link ShadowCompareScoreboardPane#getUpdatedResultsTableModel()} for further details.)
+                if (rank.contentEquals("?") && team.contentEquals("?") && numSolved.contentEquals("?") && time.contentEquals("?")) {
+                    hasCorrespondingRow = false;
+                }
+               
+                return hasCorrespondingRow;
+            }
+            
+            //we don't want any of the results cells to be editable
+            public boolean isCellEditable(int nRow, int nCol) {
+                return false;
+            }
+        };
+
+        // set default "centering" renderers for strings and integers in the table
+        DefaultTableCellRenderer centerRenderer = new DefaultTableCellRenderer();
+        centerRenderer.setHorizontalAlignment(SwingConstants.CENTER);
+//        resultsTable.setDefaultRenderer(String.class, centerRenderer);  //let team names (strings) be left-justified
+        resultsTable.setDefaultRenderer(Integer.class, centerRenderer);
+        
+        resultsTable.setRowSelectionAllowed(true);
+        resultsTable.setColumnSelectionAllowed(false);
+        resultsTable.setSelectionMode(ListSelectionModel.SINGLE_INTERVAL_SELECTION);
+        
+        return resultsTable;
+
+    }
+    
+    public enum ScoreboardType {PC2, REMOTE} ;
+    
+    private enum ScoreboardColumnId {RANK, TEAM, NUM_SOLVED, TOTAL_TIME, MATCH} ;
+    
+    /**
+     * Returns a {@link TableModel} containing data for the current comparisons between the PC2 shadow and the Remote CCS.
+     * 
+     * @param scoreboardType the scoreboard (PC2 or Remote) for which data should be extracted from the 
+     *                  specified {@link ShadowScoreboardRowComparison} array and used to populate the returned TableModel.
+     * @param comparedResults an array of ShadowScoreboardRowComparisons containing comparisons of corresponding rows of the
+     *          pc2 and remote scoreboards.
+     * 
+     * @return a {@link TableModel} populated with table information for the specified scoreboard.
+     */
+    private TableModel getUpdatedResultsTableModel(ScoreboardType scoreboardType, ShadowScoreboardRowComparison[] comparedResults) {
+
+        //define the columns for the table
+        //TODO: use Enum name/values instead of hard-coded strings for columnNames
```

Comment:
fyi - I disagree that this string array should be an enum.   One, it is busy work (no real value add) Two as is this array is consistent with all the other column definitions.   Three not all lists should be enums.  Four enums should be used in switch statements and in locations where they are likely to be changed to add new enum values.

Reply:
Well, we can spend some time, sometime, discussing/debating why our opinions on this differ...  which they do.